### PR TITLE
refactor(core): rename ngPipeDef to ɵpipe

### PR DESCRIPTION
**Pull request by kara**
Blocked on #33110 and #33116. Review last commit only.

Pipe defs are not considered public API, so the property
that contains them should be prefixed with Angular's marker
for "private" ('ɵ') to discourage apps from relying on def
APIs directly.

This commit adds the prefix and shortens the name from
ngPipeDef to pipe. This is because property names
cannot be minified by Uglify without turning on property
mangling (which most apps have turned off) and are thus
size-sensitive.

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
> core/todo/bundle.br increased by 16 bytes

This is kind of surprising given this PR (which LGTM btw!)

**Comment by kara** [comment]
@pkozlowski-opensource My guess is that it has to do with the ɵ prefix (not offset enough by the shortening of the name).

**Review by pkozlowski-opensource — APPROVED** [review]

File: packages/compiler/design/separate_compilation.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ For `CompilePipeSummary` the table looks like:
 |---------------------|-----------------------|
 | `type`              | implicit              |
 | `name`              | `ngModuleScope`       |
-| `pure`              | `ngPipeDef`           |
+| `pure`              | `ɵpipe`           |
```
Comment (petebacondarwin) [inline packages/compiler/design/separate_compilation.md]: The table layout is off now ;-)

**Review by petebacondarwin — APPROVED** [review]
LGTM

File: packages/compiler/design/separate_compilation.md (hunk 1 above)
Comment (kara) [reply packages/compiler/design/separate_compilation.md]: Ah, thank you! Fixed.

**Comment by kara** [comment]
[VE presubmit](https://test.corp.google.com/ui#id=OCL:274577851:BASE:274577869:1571064776497:5d928dac)

[Ivy presubmit](http://test/OCL:274577851:BASE:274578142:1571064903525:faaa9f4c)

**Review by IgorMinar — APPROVED** [review]

**Review by mhevery — APPROVED** [review]

**Comment by kara** [comment]
This commit is in #33142, so merging with that PR

**Pull request closed without merging**